### PR TITLE
Updated ingestion and app to handle errors more gracefully

### PR DESCRIPTION
**Pull request by jimmyo15**
Did not test it out just yet but wanted to commit so not lost.  Take a look.  I will try the streamer tomorrow with these changes.

### CODE REVIEW

**Comment by MSnowman** [comment]
Nice I’ll look through it.  At first glance looks like something I was working on last night too.  Ha.  I kicked it off today to see if it works.  Like it.  Will compare tonight.

**Comment by jimmyo15** [comment]
@MSnowman I included the S3 credential retrieval in this.  Will update once we have the full credential file put together.

**Comment by jimmyo15** [comment]
@MSnowman updated timeout logic

File: twitterApp.py [hunk 1]
```diff
@@ -5,35 +5,70 @@
...
+max_conn_issue_retries = 100
+conn_snooze_time = 30
+tcpip_snooze_increment = 0.25
+tcpip_error_count = 0
 
 #Stream Tweets
-#stream.statuses.filter(track=track_terms)
+while True:
+    if conn_issue_retries > max_conn_issue_retries:
+        break
```
Comment (MSnowman) [inline twitterApp.py]: `print('Streamer has reached the maximum connection retries and been terminated')`

Comment (MSnowman) [reply twitterApp.py]: Add the code above line 52

File: twitterApp.py [hunk 2]
```diff
@@ -5,35 +5,70 @@
...
+    try:
+        stream.statuses.filter(track=track_terms)
+    except (Timeout, ssl.SSLError) as exc:
+        if isinstance(exc, ssl.SSLError):
```
Comment (MSnowman) [inline twitterApp.py]: add the following to exceptions? ReadTimeoutError, ConnectionError
ConnectionError is in requests.exceptions
ReadTimeoutError is in urllib3.exceptions

File: twitterApp.py [hunk 3]
```diff
@@ -5,35 +5,70 @@
...
+track_terms = ['abcdefg']
+
+conn_issue_retries = 0
+max_conn_issue_retries = 100
+conn_snooze_time = 30
+tcpip_snooze_increment = 0.25
+tcpip_error_count = 0
 
 #Stream Tweets
-#stream.statuses.filter(track=track_terms)
```
Comment (MSnowman) [inline twitterApp.py]: I think we should make this a function.  This way we can call it with whatever terms we want.  Will help us get away from hardcoding

def start_streaming(track_terms, language=['en']):

**Pull request merged**